File: MyWebExtension/src/Services/Interfaces/IReqService.ts
import { StaffDirectoryItems } from "@docsvision/webclient/BackOffice/StaffDirectoryItems";
import { serviceName } from "@docsvision/webclient/System/ServiceUtils"
import { ChangeDirPhoneInfoRequest } from "../../Models/IChangeDirPhoneInfoRequest";
import { IChangeCityInfoRequest } from "../../Models/IChangeCityInfoRequest";
import { IChangeCityInfoResponse } from "../../Models/IChangeCityInfoResponse";
import { ApiResponse } from "../../Models/IApiResponse";
import { FlightDataRequest } from "../../Models/IFlightDataRequest";

export interface IReqService {
    GetReqName(documentId: string): Promise<string>
    ChangeDirPhoneInfo(request: ChangeDirPhoneInfoRequest): Promise<Boolean>
    ChangeCityInfo(request: IChangeCityInfoRequest): Promise<IChangeCityInfoResponse>
    GetCheapTickets(request: FlightDataRequest): Promise<ApiResponse> 
}

export type $ReqService = { activityPlanService: IReqService }; 
export const $ReqService = serviceName<$ReqService, IReqService>(x => x.activityPlanService); 

File: MyServerExtension/Services/IReqService.cs
using DocsVision.Platform.WebClient;
using ServerExtension.Models;
using System;
using System.Threading.Tasks;

namespace ServerExtension.Services
{
	public interface IReqService {
		string GetReqName(SessionContext sessionContext, Guid cardId);
        bool ChangeDirPhoneInfo(SessionContext sessionContext, IChangeDirPhoneInfoRequest data);
        //IChangeCityInfoResponse ChangeCityInfo(SessionContext sessionContext, IChangeCityInfoRequest data);
        void InitReq(SessionContext sessionContext, Guid cardId);
        ApiResponse GetCheapTickets(SessionContext sessionContext, IFlightDataRequest data);
    }
}
